WIP: sync code countdown on Android

AlexeyBarabash opened the pull request:
<!-- Add brave-browser issue below that this PR will resolve -->
Resolves

<!-- CI-related labels that can be applied to this PR:
* CI/run-audit-deps (1) - check for known npm/cargo vulnerabilities (audit_deps)
* CI/run-network-audit (1) - run network-audit
* CI/run-upstream-tests - run Chromium unit and browser tests on Linux and Windows (otherwise only on Linux)
* CI/run-linux-arm64, CI/run-macos-arm64, CI/run-windows-arm64, CI/run-windows-x86 - run builds that would otherwise be skipped
* CI/skip - do not run CI builds (except noplatform)
* CI/skip-linux-x64, CI/skip-android, CI/skip-macos-x64, CI/skip-ios, CI/skip-windows-x64 - skip CI builds for specific platforms
* CI/skip-upstream-tests - do not run Chromium unit, or browser tests (otherwise only on Linux)
* CI/skip-all-linters - do not run presubmit and lint checks
* CI/storybook-url (1) - deploy storybook and provide a unique URL for each build

(1) applied automatically when some files are changed (see: https://github.com/brave/brave-core/blob/master/.github/labeler.yml)
-->

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

SergeyZhukovsky commented on android/java/res/drawable/brave_sync_toggle_on_btn.xml:
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) 2019 The Brave Authors. All rights reserved.
```
2024

AlexeyBarabash replied:
thanks, fixed at https://github.com/brave/brave-core/pull/23051/commits/05e691716ccc6aa01fd73d6e7e868030614909b3

SergeyZhukovsky commented on android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java:
```diff
@@ -719,9 +751,101 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
+                                assert codePhrase != null && !codePhrase.isEmpty();
+                                String timeLimitedWords =
+                                        getBraveSyncWorker()
+                                                .getTimeLimitedWordsFromPure(codePhrase);
```
do we need to do that native on UI thread? I would do it before the `runOnUiThread`

AlexeyBarabash replied:
I can't explain why I posted that into UI thread, as it already was in UI. Thanks for finding, fixed at https://github.com/brave/brave-core/pull/23051/commits/43cafddcff2013f279118eb258965f7406b4b3cb .

SergeyZhukovsky commented on android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java:
```diff
@@ -719,9 +751,101 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
+                                LocalDateTime notAfterTime =
+                                        getBraveSyncWorker()
+                                                .getNotAfterFromFromTimeLimitedWords(
+                                                        timeLimitedWords);
```
the same as the above

AlexeyBarabash replied:
the same, https://github.com/brave/brave-core/commit/43cafddcff2013f279118eb258965f7406b4b3cb

SergeyZhukovsky commented on android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java:
```diff
@@ -719,9 +751,101 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
+                            @Override
+                            public void run() {
+                                String seedHex =
+                                        getBraveSyncWorker().getSeedHexFromWords(getPureWords());
```
the same about the native on UI thread

AlexeyBarabash replied:
and also here, https://github.com/brave/brave-core/commit/43cafddcff2013f279118eb258965f7406b4b3cb

AlexeyBarabash commented on android/java/res/drawable/sync_code_countdown_bg.xml:
```diff
@@ -0,0 +1,12 @@
+-->
+
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="#E2F0FF"/>
```
:+1:  fixed at https://github.com/brave/brave-core/pull/23051/commits/97bdae547f6b09a4a6e89f271e57fc1e0c9212c9

AlexeyBarabash commented on components/brave_sync/time_limited_words.cc:
```diff
@@ -210,4 +211,30 @@ std::string TimeLimitedWords::GenerateResultToText(
+      time_limited_words, " ", base::WhitespaceHandling::TRIM_WHITESPACE,
+      base::SplitResult::SPLIT_WANT_NONEMPTY);
+  size_t num_words = words.size();
+  DCHECK_EQ(num_words, kWordsV2Count);
```
thanks, fixed https://github.com/brave/brave-core/pull/23051/commits/94ad34f39a1d46762872716149c389148a366a22 , that DCHECK is applied against own generated words, but nevertheless I removed it.

what did you mean at 
> test coverage for not 25 words.

?

darkdh replied:
Simple unit test like calling `GetNotAfter` with 24 words and we should get empty time

AlexeyBarabash commented on components/brave_sync/time_limited_words.cc:
```diff
@@ -210,4 +211,30 @@ std::string TimeLimitedWords::GenerateResultToText(
+  }
+
+  int days_encoded = GetIndexByWord(words[kWordsV2Count - 1]);
+  // +36h is the furthest time to satisfy thelse lines from
```
thanks, fixed https://github.com/brave/brave-core/pull/23051/commits/39add543259cf28ebf61f1c7c229853b4c59d7d7

AlexeyBarabash commented on components/brave_sync/time_limited_words.cc:
```diff
@@ -210,4 +211,30 @@ std::string TimeLimitedWords::GenerateResultToText(
+
+  int days_encoded = GetIndexByWord(words[kWordsV2Count - 1]);
+  // +36h is the furthest time to satisfy thelse lines from
+  // TimeLimitedWords::ParseImpl:
```
I made here `TimeLimitedWords::GetNotAfter` use `ParseImpl`
https://github.com/brave/brave-core/pull/23051/commits/9cf9434e7d28758bbdb90d7c73e50aa3b7c6a262  just to illustrate, 

The result is still +36 hours, and I'd rather not to merge it.

SergeyZhukovsky reviewed (approved):
++

mkarolin reviewed (approved):
`strings`++

darkdh commented on components/brave_sync/time_limited_words.cc:
```diff
@@ -210,4 +212,35 @@ std::string TimeLimitedWords::GenerateResultToText(
+  // Iterate with 1 hour interval to find the point when the words will be
+  // rejected
+  for (base::Time time_to_check = anchorTime;;
+       time_to_check += base::Hours(1)) {
```
This loop is no better than the original version.
I think it would be easy to understand and clear to use same `GetRoundedDaysDiff` with original logic rather than hard coded 36hr which needs further explanation.

darkdh replied:
Probably something like `DCHECK_EQ(GetRoundedDaysDiff(anchor_time + base::Hours(36), anchor_time), 2));`
and declare 36h as constant should be enough.

AlexeyBarabash replied:
I reverted the looped version and tried to put the best description I could make. DCHECKs are also there.

https://github.com/brave/brave-core/pull/23051/commits/b03e68db97fe0fe622952b80d41e00062ee4046c

thanks for the advises, tips and discussion :+1:

thypon commented on components/brave_sync/time_limited_words.cc:
```diff
@@ -210,4 +212,35 @@ std::string TimeLimitedWords::GenerateResultToText(
+    return base::Time();
+  }
+
+  int days_encoded = GetIndexByWord(words[kWordsV2Count - 1]);
```
unsigned to signed @AlexeyBarabash  ?

AlexeyBarabash replied:
@thypon 
I don't really understand why did  this alert appear.

I could not find the exact rule from  https://github.com/brave/security-action/blob/main/assets/semgrep_rules/client/signed-unsigned-conversion.yaml by logs https://github.com/brave/brave-core/actions/runs/8822415566/job/24220547524?pr=23051

Looked on the code:

https://github.com/brave/brave-core/pull/23051/commits/9cf9434e7d28758bbdb90d7c73e50aa3b7c6a262#diff-3c1466540e68a4bd6e339da81995420c095e6044ce7f699fb7e4311b435039c9R56:

```int TimeLimitedWords::GetIndexByWord(const std::string& word) {```

then ```int days_encoded``` also is signed

maybe `kWordsV2Count - 1` ?  but this is a positive constant.
In anyway changed at https://github.com/brave/brave-core/pull/23051/commits/cdf08d967f56855084c69c0f6cc0999720c3e9f9 to try if it will help.

thypon commented on components/brave_sync/time_limited_words.cc:
```diff
@@ -210,4 +211,41 @@ std::string TimeLimitedWords::GenerateResultToText(
+    return base::Time();
+  }
+
+  int days_encoded = GetIndexByWord(words[kWordsV2Count - 1u]);
```
False positive :/

AlexeyBarabash replied:
:+1:

AlexeyBarabash replied:
@thypon 
yeah, that would be a problem if `kWordsV2Count` would be a variable

AlexeyBarabash replied:
... in addition there is an array size check above 
```
  size_t num_words = words.size();

  if (num_words != kWordsV2Count) {
    return base::Time();
  }
```

darkdh reviewed (approved):
++

deeppandya commented on android/java/res/drawable/brave_sync_toggle_off_btn.xml:
```diff
@@ -7,14 +7,5 @@
 
 <shape xmlns:android="http://schemas.android.com/apk/res/android"
```
this drawable is only being used for the transparent bg ? can we use transparent bg instead ?

AlexeyBarabash replied:
yes, indeed.
fixed at https://github.com/brave/brave-core/pull/23051/commits/3c79b2b8cdbd58c4ff1dda3ea63e4529942b74a4

deeppandya commented on android/java/res/layout/brave_sync_add_mobile.xml:
```diff
@@ -80,14 +80,32 @@
+            android:visibility="gone"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="Generate New Code"
```
Should we translate the text here ?

AlexeyBarabash replied:
Yes, my bad, added string to resource but didn't update the button.
Thanks, fixed at https://github.com/brave/brave-core/pull/23051/commits/2947f2e3e1138052304f695dc2be374d90aa540f

deeppandya commented on android/java/res/layout/fragment_sync_code_countdown.xml:
```diff
@@ -0,0 +1,77 @@
+            android:id="@+id/brave_sync_count_down_text"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:layout_marginEnd="8dp"
+            android:layout_marginTop="8dp"
+            android:layout_marginBottom="8dp"
```
```suggestion
            android:layout_margin="8dp"
```

deeppandya commented on android/java/res/layout/fragment_sync_code_countdown.xml:
```diff
@@ -0,0 +1,77 @@
+            android:layout_height="wrap_content"
+            android:contentDescription="@null"
+            app:srcCompat="@drawable/brave_countdown_timer"
+            android:layout_marginStart="8dp"
+            android:layout_marginEnd="8dp"
+            android:layout_marginTop="8dp"
+            android:layout_marginBottom="8dp" />
```
```suggestion
            android:layout_margin="8dp" />
```

deeppandya commented on android/java/res/layout/fragment_sync_code_countdown.xml:
```diff
@@ -0,0 +1,77 @@
+            android:layout_height="wrap_content"
+            android:contentDescription="@null"
+            app:srcCompat="@drawable/warning_circle_filled"
+            android:layout_marginStart="8dp"
+            android:layout_marginEnd="8dp"
+            android:layout_marginTop="8dp"
+            android:layout_marginBottom="8dp" />
```
```suggestion
            android:layout_margin="8dp" />
```

deeppandya commented on android/java/res/layout/fragment_sync_code_countdown.xml:
```diff
@@ -0,0 +1,77 @@
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:layout_marginEnd="8dp"
+            android:layout_marginTop="8dp"
+            android:layout_marginBottom="8dp"
```
```suggestion
            android:layout_margin="8dp"
```

deeppandya commented on android/java/res/layout/brave_sync_add_laptop.xml:
```diff
@@ -64,36 +64,59 @@
-            android:textColor="@color/brave_theme_color" />
+            android:layout_height="0dp"
+            android:layout_weight="1"
+            android:layout_marginTop="15dp"
+            android:layout_marginStart="15dp"
+            android:layout_marginEnd="15dp"
+            android:layout_marginBottom="15dp" />
```
```suggestion
            android:layout_margin="15dp" />
```

AlexeyBarabash commented on android/java/res/drawable/brave_sync_btn.xml:
```diff
@@ -8,11 +8,11 @@
-        android:bottomLeftRadius="30dp"
-        android:topRightRadius="30dp"
-        android:topLeftRadius="30dp" />
+    <corners android:bottomRightRadius="12dp"
```
thanks, fixed at https://github.com/brave/brave-core/pull/23051/commits/f68bda9f5c255f155a15d9b798a8ec829621b030

AlexeyBarabash commented on android/java/res/drawable/brave_sync_btn_transparent.xml:
```diff
@@ -1,20 +1,16 @@
-    <stroke android:width="1px" android:color="@color/brave_theme_color" />
-
+    <stroke android:width="1dip" android:color="@color/brave_sync_main_button_color"/>
+    <corners android:bottomRightRadius="12dp"
```
also fixed at https://github.com/brave/brave-core/pull/23051/commits/f68bda9f5c255f155a15d9b798a8ec829621b030

AlexeyBarabash commented on android/java/res/values/brave_colors.xml:
```diff
@@ -89,6 +89,17 @@
     <color name="brave_sync_bg_color">@android:color/transparent</color>
     <color name="brave_sync_border_color">@color/modern_grey_50</color>
     <color name="brave_sync_bg_secondary_color">@android:color/white</color>
+    <color name="brave_sync_stop_watch_icon_color">#0061AA</color>
```
I did some changes at https://github.com/brave/brave-core/pull/23051/commits/bf304b5e929992aa5ab3a868b61051c2cf858731

I had top copy values from `src/brave/node_modules/@brave/leo/tokens/android/values/colors.xml`, because direct attempt to include colors
```
--- a/android/brave_java_resources.gni
+++ b/android/brave_java_resources.gni
@@ -963,6 +963,7 @@ brave_java_resources = [
   "java/res/values/array.xml",
   "java/res/values/brave_attrs.xml",
   "java/res/values/brave_colors.xml",
+  "//brave/node_modules/@brave/leo/tokens/android/values/colors.xml",
   "java/res/values/brave_dimens.xml",
```

didn't work :
```
[1/2849] ACTION //chrome/android:chrome_app_java_resources(//build/toolchain/android:android_clang_arm)
FAILED: obj/chrome/android/chrome_app_java_resources.resources.zip obj/chrome/android/chrome_app_java_resources.resources.zip.info gen/chrome/android/chrome_app_java_resources_R.txt 
python3 ../../build/android/gyp/prepare_resources.py --depfile gen/chrome/android/chrome_app_java_resources.d --res-sources-path=gen/chrome/android/chrome_app_java_resources.res.sources --resource-zip-out obj/chrome/android/chrome_app_java_resources.resources.zip --r-text-out gen/chrome/android/chrome_app_java_resources_R.txt
Error: Found files not listed in the sources list of the BUILD.gn target:
../../brave/node_modules/@brave/leo/tokens/android/drawable/ic_browser_mobile_tab_closes_app.xml
...
a lot of other files
...
```

And it looks this is the task beyond this PR.
wdyt?

deeppandya replied:
@AlexeyBarabash we need to copy/paste colors at the moment into brave_colors.xml as we don't use all of the colors mentioned in the generated file. i am going to replace existing colors with generated tokens soon.

AlexeyBarabash replied:
gotcha, thanks

deeppandya reviewed (approved):
LGTM

aguscruiz reviewed (changes requested):
Added design feedback in this page
https://www.figma.com/file/GXnHuBCVRY397UNJyvu44q/Sync-V2?type=design&node-id=483-1279&mode=design

AlexeyBarabash commented:
> Added design feedback in this page https://www.figma.com/file/GXnHuBCVRY397UNJyvu44q/Sync-V2?type=design&node-id=483-1279&mode=design

Hi @aguscruiz , could you please look again on the updated layouts, the video is at [the first message at the PR](https://github.com/brave/brave-core/pull/23051#issue-2240166626)

aguscruiz commented:
I noticed that here, when you scroll dow, the tabs get hidden out of scroll. They should be sticky to the top so you can always see the tab 

See reference: https://m3.material.io/components/tabs/guidelines#bcd3709b-14f1-4ac9-b7f7-4bc786e2a8eb

<img width="325" alt="image" src="https://github.com/brave/brave-core/assets/15734630/89587ef5-a8cd-401e-a905-e6583f075991">

aguscruiz commented:
One more, I forgot that you don't have the design system colors setup automatically for light/dark modes, sorry. 

I added the dark mode version of the alerts and texts, since in that video you're using light mode's for both.

Timer:
https://www.figma.com/design/GXnHuBCVRY397UNJyvu44q/Sync-V2?node-id=658-11632&t=4q3vxGxTm9k7zMUS-1

Error message:
https://www.figma.com/design/GXnHuBCVRY397UNJyvu44q/Sync-V2?node-id=658-11770&t=4q3vxGxTm9k7zMUS-1

AlexeyBarabash commented:
Hey @aguscruiz , 
I have fixed scroll and updated colors for Dark theme according to the latest Figma designs. The updated video is at the first message here https://github.com/brave/brave-core/pull/23051#issue-2240166626. Could you please look again?

AlexeyBarabash commented:
sorry, I just found one wrong color for Dark theme, at `Generate New Code` font. fixing.

AlexeyBarabash commented:
Fixed at https://github.com/brave/brave-core/pull/23051/commits/0345b65bd22d3b82310eb186a3dd812836b7ab3d  .
Updated video. @aguscruiz PR is ready for review again.

aguscruiz reviewed (commented):
Looks great 👌

SergeyZhukovsky commented on browser/ui/android/strings/android_brave_strings.grd:
```diff
@@ -1161,24 +1158,18 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
-      <message name="IDS_BRAVE_SYNC_ADD_LAPTOP_TEXT_PART_1" desc="Text for Brave Sync add a laptop description part 1.">
-	On your computer, navigate to Brave Settings > Sync. Choose "Enter a Sync Chain Code".
+      <message name="IDS_BRAVE_SYNC_ADD_LAPTOP_TEXT" desc="Text for Brave Sync add a laptop description.">
+  On your computer, navigate to Brave Settings > Sync. Choose "Enter a Sync Chain Code".\n\nOn your target computer, navigate to Brave Sync in settings and click the button "I have a Sync Code".\nEnter the sync chain code words shown below.
```
could you format strings to follow the common pattern we have?

AlexeyBarabash replied:
yes, thanks, fixed at https://github.com/brave/brave-core/pull/23051/commits/1278dd325c3ec899fdec212b6cbcf5c3b6c44cca

SergeyZhukovsky reviewed (approved):
++

deeppandya commented on android/brave_java_resources.gni:
```diff
@@ -742,8 +745,10 @@ brave_java_resources = [
+  "java/res/drawable/warning_circle_filled_theme_color.xml",
   "java/res/drawable/web_notification_small_icon_background.xml",
   "java/res/drawable/white_rounded_holo_button_thin.xml",
+  "java/res/font/roboto_mono_regular.ttf",
```
@aguscruiz do we need new font for the changes ?

AlexeyBarabash replied:
@deeppandya 
 I believe, we need. I could specify Roboto regular, it present on devcies, but we need `mono` which means the width of all chars is equal. I couldn't find the way to enable roboto mono without embeding.
 
 I had to take it from https://fonts.google.com/specimen/Roboto+Mono

deeppandya commented on android/java/res/layout/brave_sync_add_laptop.xml:
```diff
@@ -9,7 +9,11 @@
     android:layout_height="match_parent"
-    android:fillViewport="true" >
+    android:fillViewport="true"
+    android:layout_marginStart="20dp"
+    android:layout_marginEnd="20dp"
```
```suggestion
    android:layout_marginStart="20dp"
    android:layout_marginEnd="20dp"
```
android:layout_marginHorizontal="20dp"

deeppandya replied:
```suggestion
   android:layout_marginHorizontal="20dp"
```

deeppandya commented on android/java/res/layout/brave_sync_add_laptop.xml:
```diff
@@ -52,48 +31,86 @@
+            android:layout_height="wrap_content"
             android:paddingStart="15dip"
-            android:paddingEnd="15dip" >
+            android:paddingEnd="15dip"
```
nit :  you can use android:paddingHorizontal="15dp"

AlexeyBarabash replied:
thanks fixed at 309e389c8db4ebc9f653667a344389dfb089b834

deeppandya commented on android/java/res/layout/fragment_sync_code_countdown.xml:
```diff
@@ -0,0 +1,67 @@
+            android:layout_margin="8dp"
+            android:textColor="@color/systemfeedback_info_text"
+            android:gravity="center_vertical"
+            android:textSize="14sp"
```
i think we should use style from nala here.

deeppandya commented on android/java/res/layout/fragment_sync_code_countdown.xml:
```diff
@@ -0,0 +1,67 @@
+            android:text="@string/brave_sync_code_expired_block_text"
+            android:textColor="@color/systemfeedback_error_text"
+            android:gravity="center_vertical"
+            android:textSize="14sp"
```
we should use nala style here.

AlexeyBarabash commented:
@deeppandya , @aguscruiz 
Replaced icons from Nala (`src/brave/node_modules/@brave/leo/tokens/android/drawable`) at https://github.com/brave/brave-core/pull/23051/commits/f117b50ab492839f394c720ca02a0aba303398ee.

To support the Dark Theme I had to update colors.

`ic_timer.xml`:
![image](https://github.com/brave/brave-core/assets/24739341/05e78de3-9ae2-47c3-b032-ddbc3c2700f2)

The icon `ic_warning_circle_filled.xml` already was in resources and it was different from the one from `@brave/leo/tokens`:

![image](https://github.com/brave/brave-core/assets/24739341/c8ec6b37-a999-4ff2-b0da-df8a28b60851)

so I had to use `ic_warning_circle_filled2.xml` name to avoid conflicts at other places:
```
			src/brave/android/java/org/chromium/chrome/browser/custom_layout/PasswordStrengthMeterView.java
			src/brave/android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java
			src/brave/android/java/res/layout/fragment_sign_message_error.xml
			src/brave/android/java/res/layout/rewards_onboarding_error.xml
```

Also I updated the value of the color:
![image](https://github.com/brave/brave-core/assets/24739341/97d5d306-3735-416e-a55e-2e24198baa50)

deeppandya commented:
@AlexeyBarabash changes look good to me. @aguscruiz can you confirm if we need Roboto-mono font ? it adds 83 kbs in the package.

aguscruiz commented:
IT's ok to use the regular roboto for now, if we end up needing it on other places we'll have to add it. But if this is the only place, then let's use regular Roboto

AlexeyBarabash commented:
> @AlexeyBarabash changes look good to me. @aguscruiz can you confirm if we need Roboto-mono font ? it adds 83 kbs in the package.

I can use default monospace font to avoid 83kb size impact
Here is how it looks
default|roboto-mono
---|---
![Screenshot_20240619-223123](https://github.com/brave/brave-core/assets/24739341/4842b6a6-7377-410e-b4bc-eedd07c244f8)|![Screenshot_20240619-221944 (1)](https://github.com/brave/brave-core/assets/24739341/7a2ca7bc-fce5-4c11-8b5c-111c3fdaaf87)

by specifying `<item name="android:typeface">monospace</item>`  at style.
It doesn't look the same, but at least has the same width of chars.

Maybe it would be better to use this option?

aguscruiz commented:
That's perfect 👌

AlexeyBarabash commented:
removed `roboto_mono_regular.ttf` and rebased
